File: iosApp/iosApp/Features/JobOffer/JobOfferStatsCard.swift
//
//  JobOfferStatsCard.swift
//  iosApp
//
//  Created by DAN BIZWA on 19/09/2026.
//

// iosApp/iosApp/Features/JobOffer/JobOfferStatsCard.swift
import SwiftUI
import SharedLogic

// Miroir de androidApp/.../ui/joboffer/JobOfferStatsCard.kt.
struct JobOfferStatsCard: View {
    let offers: [JobOffer]
    @Environment(\.colorScheme) private var colorScheme

    private var counts: [(ApplicationStatus, Int)] {
        ApplicationStatus.entries.compactMap { status in
            let count = offers.filter { $0.status == status }.count
            return count > 0 ? (status, count) : nil
        }
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            // Total + libellé lus ensemble par VoiceOver (« 3 applications tracked » / « 3 candidatures suivies »)
            VStack(alignment: .leading, spacing: 0) {
                // headlineSmall.copy(fontSize = 34.sp, Bold) — le lineHeight reste celui de headlineSmall (30sp)
                Text("\(offers.count)")
                    .appTextStyle(.headlineSmall.copy(weight: .bold, size: 34), fixedLineHeight: true)
                    .foregroundStyle(Color.tealOnContainer)
                Text(L(offers.count > 1 ? "stats_tracked_other" : "stats_tracked_one"))
                    .appTextStyle(.bodyMedium)
                    .foregroundStyle(Color.tealOnContainer.opacity(0.8))
            }
            .accessibilityElement(children: .combine)

            if !counts.isEmpty {
                // FlowLayout (équivalent FlowRow) : 4-5 badges ne tiennent pas sur une ligne d'iPhone / à grande taille de texte
                FlowLayout(spacing: 8) {
                    ForEach(counts, id: \.0) { status, count in
                        Text(status.shortLabel(count: count))
                            .appTextStyle(.labelLarge.copy(weight: .semiBold))
                            .foregroundStyle(status.color)
                            .padding(.horizontal, 10)
                            .padding(.vertical, 6)
                            .background(status.color.opacity(Color.statusBadgeTintAlpha(for: colorScheme)))
                            .clipShape(RoundedRectangle(cornerRadius: 10))
                    }
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(.top, 16)
            }
        }
        .padding(20)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(Color.tealContainer)                  // primaryContainer
        .clipShape(RoundedRectangle(cornerRadius: 12))    // Card : shapes.medium (était 16)
    }
}

/*
#Preview {
    JobOfferStatsCard()
}*/

File: androidApp/src/main/kotlin/com/dmb/joblog/ui/joboffer/JobOfferStatsCard.kt
package com.dmb.joblog.ui.joboffer

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.dmb.joblog.domain.model.ApplicationStatus
import com.dmb.joblog.domain.model.JobOffer
import com.dmb.joblog.ui.theme.LocalStatusPalette
import com.dmb.joblog.ui.theme.color

import androidx.compose.foundation.shape.RoundedCornerShape

import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp
import androidx.compose.ui.res.stringResource
import com.dmb.joblog.R

@OptIn(ExperimentalLayoutApi::class)
@Composable
fun JobOfferStatsCard(offers: List<JobOffer>) {
    val total = offers.size
    val counts = ApplicationStatus.entries.associateWith { status ->
        offers.count { it.status == status }
    }.filterValues { it > 0 }

    Card(
        modifier = Modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.primaryContainer
        ),
        elevation = CardDefaults.cardElevation(defaultElevation = 0.dp)
    ) {
        Column(modifier = Modifier.padding(20.dp)) {
            // Total + libellé lus ensemble (« 3 applications tracked » / « 3 candidatures suivies »)
            Column(modifier = Modifier.semantics(mergeDescendants = true) {}) {
                Text(
                    "$total",
                    style = MaterialTheme.typography.headlineSmall.copy(fontSize = 34.sp, fontWeight = FontWeight.Bold),
                    color = MaterialTheme.colorScheme.onPrimaryContainer
                )
                Text(
                    stringResource(if (total > 1) R.string.stats_tracked_other else R.string.stats_tracked_one),
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onPrimaryContainer.copy(alpha = 0.8f)
                )
            }

            if (counts.isNotEmpty()) {
                Spacer(modifier = Modifier.height(16.dp))
                // FlowRow : avec 4-5 statuts, une Row simple déborde de l'écran (360 dp) ou à grande taille de police
                FlowRow(
                    horizontalArrangement = Arrangement.spacedBy(8.dp),
                    verticalArrangement = Arrangement.spacedBy(8.dp),
                    modifier = Modifier.fillMaxWidth()
                ) {
                    counts.forEach { (status, count) ->
                        StatusBadge(status = status, count = count)
                    }
                }
            }
        }
    }
}

@Composable
private fun StatusBadge(status: ApplicationStatus, count: Int) {
    Box(
        modifier = Modifier
            .background(status.color().copy(alpha = LocalStatusPalette.current.badgeTintAlpha), RoundedCornerShape(10.dp))
            .padding(horizontal = 10.dp, vertical = 6.dp)
    ) {
        Text(
            status.shortLabel(count),
            style = MaterialTheme.typography.labelLarge.copy(fontWeight = FontWeight.SemiBold),
            color = status.color()
        )
    }
}
